### PR TITLE
(TWILL-136) Override equals and hashCode for JvmOptions.DebugOptions to test equality

### PR DESCRIPTION
**Pull request by hsaputra**
The current code for JvmOptions.DebugOptions does not have equals and hashCode overriden for equality test.

This would cause fail comparison for DebugOptions.NO_DEBUG when being used in YarnTwillPreparer:

```
final class YarnTwillPreparer implements TwillPreparer {

...

  @Override
  public TwillPreparer enableDebugging(boolean doSuspend, String... runnables) {
    this.debugOptions = new JvmOptions.DebugOptions(true, doSuspend, ImmutableSet.copyOf(runnables));
    return this;
  }

....
  private void saveJvmOptions(Map<String, LocalFile> localFiles) throws IOException {
    if ((extraOptions == null || extraOptions.isEmpty()) &&
      JvmOptions.DebugOptions.NO_DEBUG.equals(this.debugOptions)) {
      // If no vm options, no need to localize the file.
      return;
    }
    ...
  }
...
}
```

### CODE REVIEW

**Comment by hsaputra** [comment]
Update PR based on @keith-turner review comment in #42

**Comment by hsaputra** [comment]
CC @chtyim, @anew for review

File: twill-core/src/main/java/org/apache/twill/internal/JvmOptions.java [hunk 1]
```diff
@@ -83,5 +84,36 @@ public boolean doSuspend() {
...
+      if (this == object) {
+        return true;
+      }
+      if (!(object instanceof DebugOptions)) {
+        return false;
+      }
+
+      DebugOptions that = (DebugOptions) object;
+      return (this.doDebug == that.doDebug()) && (this.doSuspend == this.doSuspend()) &&
+        (this.runnables != null ? ((that.getRunnables() != null) && this.getRunnables().equals(that.getRunnables())) :
```
Comment (anew) [inline twill-core/src/main/java/org/apache/twill/internal/JvmOptions.java]: I prefer using Objects.equal(), it is much nicer to read.

File: twill-core/src/main/java/org/apache/twill/internal/JvmOptions.java [hunk 2]
```diff
@@ -83,5 +84,36 @@ public boolean doSuspend() {
...
+          (that.getRunnables() == null));
+    }
+
+    @Override
+    public int hashCode() {
+      int hash = 17;
```
Comment (anew) [inline twill-core/src/main/java/org/apache/twill/internal/JvmOptions.java]: same here, you could use Objects.hashCode()

**Comment by hsaputra** [comment]
Updated the PR per @anew review

File: twill-core/src/main/java/org/apache/twill/internal/JvmOptions.java (hunk 2 above)
Comment (anew) [inline twill-core/src/main/java/org/apache/twill/internal/JvmOptions.java]: Oh, I meant Objects.hash(...). But it's ok here.

**Comment by anew** [comment]
+1

**Comment by chtyim** [comment]
LGTM. Can you rebase to one commit?

**Comment by hsaputra** [comment]
Thanks guys, just rebased them into one commit

**Pull request closed without merging**